### PR TITLE
Feature/tao 3370 limit input length

### PR DESCRIPTION
**Pull request by ssipasseuth**
extract the existing pattern mask management from extended text interaction and apply it to both string interactions.

works better with https://github.com/oat-sa/tao-core/pull/1085 : the tooltip will display better.

### CODE REVIEW

File: views/js/qtiCommonRenderer/tpl/interactions/textEntryInteraction-counter.tpl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<span class="qti-textEntryInteraction-counter">
```
Comment (no-chris) [inline views/js/qtiCommonRenderer/tpl/interactions/textEntryInteraction-counter.tpl]: is this still used ?

Comment (ssipasseuth) [inline views/js/qtiCommonRenderer/tpl/interactions/textEntryInteraction-counter.tpl]: no indeed, nice catch !

File: views/js/test/qtiCommonRenderer/interactions/textEntry/test.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                assert.equal(getTooltipContent($input), '5/5', 'the warning message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'warning tooltip is visible');
+                assert.ok($input.hasClass('maxed'), 'has state maxed');
+
+                $input.val('1234');
+                $input.keyup();
+                assert.equal(getTooltipContent($input), '4/5', 'the instruction message is correct');
+                assert.ok(getTooltip($input).is(':visible'), 'info tooltip is visible');
+                assert.ok(!$input.hasClass('maxed'), 'has state maxed removed');
+
```
Comment (ssipasseuth) [inline views/js/test/qtiCommonRenderer/interactions/textEntry/test.js]: Interesting yes, but unfortunately, I didn't find a way to do this. Indeed if we simulate a key input event in an input, the event is fired but the input value is not updated so no way to really simulate a use typing one by one, which is a pity.

**Comment by ssipasseuth** [comment]
Ok, changed applied, please check again.

**Pull request merged**